pkg/common-controller/snapshot_controller_base.go

```diff
@@ -317,6 +317,7 @@ func (ctrl *csiSnapshotCommonController) syncContentByKey(key string) error {
 
 // checkAndUpdateSnapshotClass gets the VolumeSnapshotClass from VolumeSnapshot. If it is not set,
 // gets it from default VolumeSnapshotClass and sets it.
+// On error, it returns the original snapshot.
```

Comment:
Can you change to the following?

On error, it must return the original snapshot, not nil, because the caller syncContentByKey needs to check snapshot's timestamp.

Reply:
Done